Source/Core/Core/HW/Memmap.cpp

```diff
@@ -409,7 +409,7 @@ u8* MemoryManager::GetPointerForRange(u32 address, size_t size) const
 
   // Check that the beginning and end of the range are valid
   u8* pointer = GetPointer(address);
-  if (!pointer || !GetPointer(address + u32(size) - 1))
+  if (!pointer || (size != 0 && !GetPointer(address + u32(size) - 1)))
```

Comment:
I think this would be a bit more obvious as

```suggestion
  if (pointer == nullptr || (size != 0 && (GetPointer(address + u32(size) - 1) == nullptr)))
```